packages/vitest/src/node/stdin.ts

```diff
@@ -154,3 +166,102 @@ export function registerConsoleShortcuts(ctx: Vitest) {
     off()
   }
 }
+
+  type FilterFunc = (str: string) => Promise<string[]>
+
+async function createFilter(message: string, initial: string, filterFunc: FilterFunc) {
+  let currentKeyword: string | undefined
+  const filterRL = readline.createInterface({ input: process.stdin, escapeCodeTimeout: 50 })
+  readline.emitKeypressEvents(process.stdin, filterRL)
+  if (process.stdin.isTTY)
+    process.stdin.setRawMode(false)
+  const handler = filterHandler(filterFunc)
+  process.stdin.on('keypress', handler)
+
+  const { filter = '' }: { filter: string } = await prompt([{
+    name: 'filter',
+    type: 'text',
+    message,
+    initial,
+  }])
+
+  filterRL.close()
+  process.stdin.removeListener('keypress', handler)
+  if (process.stdin.isTTY)
+    process.stdin.setRawMode(false)
+
+  return filter
+
+  function restoreCursor() {
+    stdout().write(ansiEscapes.cursorTo(`? ${message} › `.length + (currentKeyword?.length || 0)))
+  }
+
+  function filterHandler(filterFunc: FilterFunc) {
+    const MAX_RESULT_COUNT = 10
+    return async function (str: string, key: any) {
+    // backspace
+      if (key.sequence === '\x7F') {
+        if (currentKeyword && currentKeyword?.length > 1)
+
+          currentKeyword = currentKeyword?.slice(0, -1)
+
+        else
+          currentKeyword = undefined
+      }
+      else if (key?.name === 'return') {
+        // reset current keyword
+        currentKeyword = undefined
+        return
+      }
+      else {
+        if (currentKeyword === undefined)
+          currentKeyword = str
+        else
+          currentKeyword += str
+      }
+
+      if (currentKeyword) {
+        const results = await filterFunc(currentKeyword)
+
+        if (results.length === 0) {
+          eraseAndPrint(`\nPattern matches no results`)
+        }
+        else {
+          if (results.length > MAX_RESULT_COUNT) {
+            eraseAndPrint(`\nPattern matches ${results.length} results`
+           + `\n${results.slice(0, MAX_RESULT_COUNT).map(result => c.dim(` › ${result}`)).join('\n')}${
+            c.dim(`\n   ...and ${results.length - MAX_RESULT_COUNT} more results`)}`)
+          }
+          else { eraseAndPrint(`\nPattern matches ${results.length} results` + `\n${results.map(result => c.dim(` › ${result}`)).join('\n')}`) }
+        }
+      }
+      else {
+        eraseAndPrint('\nPlease input filter pattern')
+      }
+      restoreCursor()
+    }
+  }
+}
+
+/**
+ * Print string and back to original cursor position
+ * @param str
+ */
+function eraseAndPrint(str: string) {
+  const lineBreasks = str.split('\n').length - 1
+
+  stdout().write(ansiEscapes.eraseDown)
```

Comment:
I see that we only use named shorthand for ansi codes, I think we can use them directly without installing the whole library

Reply:
Thank you for reviewing!
I'll remove the library!

Reply:
I've removed ansi-escapes library and use raw escape code as you suggested.